src/groupByKey.ts

```diff
@@ -1,7 +1,21 @@
 interface Item {
-  [key: string]: any;
+  [key: string]: any,
 }
 
-export function groupByKey(items, field) {
-  // write code here;
+interface GroupItem {
+  [key: string] : Item[],
+}
+
+export function groupByKey(items: Item[], field: keyof Item): GroupItem {
+  const resultObject: GroupItem = {};
+
+  items.forEach((item: Item) => {
```

Comment:
type of item will be inherited automatically

```suggestion
  items.forEach((item) => {
```